pytorch_lightning/trainer/trainer.py

```diff
@@ -1109,6 +1112,24 @@ def __attach_dataloaders(self, model, train_dataloader=None, val_dataloaders=Non
         if test_dataloaders is not None:
             model.test_dataloader = _PatchDataLoader(test_dataloaders)
 
+    def __attach_datamodule(self, model, datamodule=None):
```

Comment:
how about the dataloader is runtime overwritten in previous `__attach_dataloaders`

Reply:
Added conditionals to block this from happening 😄